arches/app/datatypes/datatypes.py

```diff
@@ -2094,14 +2094,28 @@ def pre_tile_save(self, tile, nodeid):
 
     def post_tile_save(self, tile, nodeid, request):
         ret = False
-        sql = """
+        create_sql = """
             SELECT * FROM __arches_create_resource_x_resource_relationships('%s') as t;
         """ % (
             tile.pk
         )
+        if tile.data[nodeid] and len(tile.data[nodeid]) > 0:
+            delete_sql = """
+                DELETE FROM resource_x_resource WHERE tileid = '%s' AND resourcexid NOT IN (%s);
```

Comment:
This actually just ends up deleting all the resource_x_resource records for the tile because the resourcexids have all changed before this is run. Even if it didn't delete all of the records, they would still all get deleted in `__arches_create_resource_x_resource_relationships ` here
https://github.com/archesproject/arches/blob/206a887aaa51868ea9b97336129e45996c7315bb/arches/app/models/migrations/9746_related_resource_post_save_bug.py#L39